lib/src/kbasesearchengine/search/ObjectData.java

```diff
@@ -33,7 +35,9 @@
     private final Optional<Object> parentData;
     private final Optional<Object> data;
     private final Map<String, String> keyProps;
-    
+    private final Map<String, ArrayList> highlight;
```

Comment:
Don't specify the implementation - this applies in several places in this class.